lib/ansible/executor/module_common.py

```diff
@@ -50,6 +55,61 @@
 
 # ******************************************************************************
 
+ZIPLOADER_DATA = u'''%(shebang)s
+# -*- coding: utf-8 -*-'
+import os
+import sys
+import base64
+import tempfile
+
+ZIPDATA = """%(zipdata)s"""
+
+def debug(command, zipped_mod):
+    # Okay to use __file__ here because we're running from a kept file
+    basedir = os.path.dirname(__file__)
+    if command == 'explode':
+        import zipfile
+        z = zipfile.ZipFile(zipped_mod)
+        for filename in z.namelist():
+            if filename.startswith('/'):
+                raise Exception('Something wrong with this module zip file: should not contain absolute paths')
```

Comment:
We should be raising an Exception because this won't be sent back to ansible.

The debug() functionality is something I worked on with @nitzmahone last night.  We realized that it would be harder to debug modules via ANSIBLE_KEEP_REMOTE_FILES using ziploader because what you would end up keeping is this wrapper with the zipped payload in ZIPDATA.  So the debug() function contains two things to ameliorate that.  First: If you run the wrapper file with ```explode``` as a command line arg it will take the payload and extract it into the ansible temporary directory (where the module file was uploaded).  Second: If you run the wrapper file with ```execute``` as a command line arg it will run the module files that were exploded to disk rather than the zipped payload that's contained in the wrapper.

These two commands enable you to debug a module using the following steps:
```
$ ANSIBLE_KEEP_REMOTE_FILES=1 ansible localhost -m ping -a data=crash -vvv
Using /etc/ansible/ansible.cfg as config file
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: badger
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1459605113.37-203813258173
480 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1459605113.37-203813258173480 `" )'
<127.0.0.1> PUT /var/tmp/tmpvdVWjY TO /home/badger/.ansible/tmp/ansible-tmp-1459605113.37-203813258173480/ping
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/badger
/.ansible/tmp/ansible-tmp-1459605113.37-203813258173480/ping'
An exception occurred during task execution. The full traceback is:
[...]
# Log into remote machine
$ /usr/bin/python /home/badger
/.ansible/tmp/ansible-tmp-1459605113.37-203813258173480/ping explode
Module expanded into: /home/badger/.ansible/tmp/ansible-tmp-1459605113.37-203813258173480/ansible
# Edit the files in /home/badger/.ansible/tmp/ansible-tmp-1459605113.37-203813258173480/ansible to introspect what is happening as the module runs.   Change the code to see if it will still run, etc.
$ /usr/bin/python /home/badger/.ansible/tmp/ansible-tmp-1459605113.37-203813258173480/ping execute
# The execute command runs the exploded files instead of the zipped payload.  So it runs the module with all of the changes you made in the step above.
```